src/test/crimson/seastore/test_object_data_handler.cc

```diff
@@ -829,6 +829,7 @@ TEST_P(object_data_handler_test_t, overwrite_then_read_within_transaction) {
     read(base, 4096 * 3);
     restart();
     epm->check_usage();
+    read(base, 4096);
```

Comment:
Sorry I'm not following why adding this line leads to sufficient unit test for the partial reading feature?

Another missing test case might be, as mentioned in another comment, to validate concurrent partial/full reads. This is necessary because (at least) the background and forground tasts are allowed to read the same extent at the same time.